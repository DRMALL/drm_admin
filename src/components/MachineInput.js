

import React from 'react'
import Input from 'antd/lib/input'
import getInputValue from '../actions/getInputValue'
import Upload from 'antd/lib/upload'
import { uploadImg } from '../commons/apis'
import Modal from 'antd/lib/modal'
import dispatch from '../actions/dispatch'
import Icon from 'antd/lib/icon'


export default props => {
  const { fileList, previewImage, previewVisible } = props.state.machine
  const uploadButton = (
      <div>
        <Icon type="plus" />
        <div className="ant-upload-text">上传图片</div>
      </div>
  )
    return(
      <div>
        <div className='machine-input-flex' >
          <div  className='machine-input-left'>设备名称</div>
          <Input  className='machine-input-right' onChange={ getInputValue }
                  style={{ width: 400 }} placeholder='输入设备名称'  />
        </div>
        <div className='machine-input-flex' >
          <div  className='machine-input-left'>设备编号</div>
          <Input  className='machine-input-right' onChange={ getInputValue }
                  style={{ width: 400 }} placeholder='输入设备编号' />
        </div>
        <div className='machine-input-flex' >
          <div  className='machine-input-left'>压力分类</div>
          <Input  className='machine-input-right' onChange={ getInputValue }
                  style={{ width: 400 }} placeholder='输入压力分类' />
        </div>
        <div className='machine-input-flex' >
          <div  className='machine-input-left'>燃料分类</div>
          <Input  className='machine-input-right' onChange={ getInputValue }
                  style={{ width: 400 }} placeholder='输入燃料分类' />
        </div>
        <div className='machine-input-flex' >
          <div  className='machine-input-left'>排量分类</div>
          <Input  className='machine-input-right' onChange={ getInputValue }
                  style={{ width: 400 }} placeholder='输入排量分类' />
        </div>
        <div className='machine-input-flex' >
          <div  className='machine-input-left'>设备所在地</div>
          <Input  className='machine-input-right' onChange={ getInputValue }
                  style={{ width: 400 }} placeholder='输入设备所在地' />
        </div>
        <div className='machine-input-flexs' >
          <div className='machine-input-left' >设备描述</div>
          <Input.TextArea rows={4} style={{ width: 400 }}
                  placeholder='输入设备描述'/>
        </div>
        <div>
          <div>上传设备图片</div>
          <div>
            <Upload
              action={`${uploadImg}?token=${localStorage.getItem('token')}`}
              name="news"
              listType="picture-card"
              fileList={fileList}
              multiple={ true }
              onPreview={ handlePreviewMachine }
              onChange={ handleChangeMachine }
            >
              {fileList.length >= 10 ? null : uploadButton}
            </Upload>
            <Modal visible={ previewVisible } footer={null} onCancel={ handleCancelMachine }>
              <img alt="example" style={{ width: '100%' }} src={ previewImage } />
            </Modal>
          </div>
        </div>
      </div>
      )
}

function handlePreviewMachine(e) {
  console.log(e.response.data.url)
  const { url } = e.response.data
  dispatch('MESSAGE_PREVIEW_IMG', url)
}

function handleChangeMachine(e){
  dispatch('MESSAGE_UPLOAD_IMG', e.fileList)
}

function handleCancelMachine(e){
  dispatch('MESSAGE_HIDDEN_MODAL')
}

