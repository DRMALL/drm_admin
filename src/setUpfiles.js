
//mock localStorage和react 的，只需要在文件中引入就行
import React from 'react';
if (typeof window !== 'undefined') {
    const matchMediaPolyfill = function matchMediaPolyfill() {
    return {
      matches: false,
      addListener() {
      },
      removeListener() {
      },
    };
  };
  window.matchMedia = window.matchMedia || matchMediaPolyfill;

  window.React = React;
  window.localStorage = ( function storageMock() {
    var storage = {};
    return {
      setItem: function(key, value) {
        storage[key] = value || '';
      },
      getItem: function(key) {
        return key in storage ? storage[key] : null;
      },
      removeItem: function(key) {
        delete storage[key];
      },
      get length() {
        return Object.keys(storage).length;
      },
      key: function(i) {
        var keys = Object.keys(storage);
        return keys[i] || null;
      }
    }
  })()
}
