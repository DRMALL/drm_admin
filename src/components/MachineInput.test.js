

import MachineInput from './MachineInput'
import React from 'react'
import renderer from 'react-test-renderer'
import setUpfiles from '../setUpfiles'

it('MachineInput test', ()=> {
  const state = { machine: { fileList: [] } }
  const tree = renderer.create(<MachineInput state= { state } />)
  expect(tree).toMatchSnapshot()
})
