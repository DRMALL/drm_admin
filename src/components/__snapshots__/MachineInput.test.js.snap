// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MachineInput test 1`] = `
<div>
  <div
    className="machine-input-flex"
  >
    <div
      className="machine-input-left"
    >
      设备名称
    </div>
    <input
      className="ant-input machine-input-right"
      data-id="name"
      data-path="MACHINE"
      disabled={false}
      onChange={[Function]}
      onKeyDown={[Function]}
      placeholder="输入设备名称"
      style={
        Object {
          "width": 400,
        }
      }
      type="text"
      value=""
    />
  </div>
  <div
    className="machine-input-flex"
  >
    <div
      className="machine-input-left"
    >
      设备编号
    </div>
    <input
      className="ant-input machine-input-right"
      data-id="number"
      data-path="MACHINE"
      disabled={false}
      onChange={[Function]}
      onKeyDown={[Function]}
      placeholder="输入设备编号"
      style={
        Object {
          "width": 400,
        }
      }
      type="text"
      value=""
    />
  </div>
  <div
    className="machine-input-flex"
  >
    <div
      className="machine-input-left"
    >
      压力分类
    </div>
    <div
      className="ant-select ant-select-enabled"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={undefined}
      onMouseLeave={undefined}
      onTouchStart={[Function]}
      style={
        Object {
          "width": 400,
        }
      }
    >
      <div
        aria-autocomplete="list"
        aria-expanded={false}
        aria-haspopup="true"
        className="ant-select-selection
                  ant-select-selection--single"
        onKeyDown={[Function]}
        role="combobox"
        tabIndex={0}
      >
        <div
          className="ant-select-selection__rendered"
        >
          <div
            className="ant-select-selection__placeholder"
            onClick={[Function]}
            onMouseDown={[Function]}
            style={
              Object {
                "WebkitUserSelect": "none",
                "display": "block",
                "userSelect": "none",
              }
            }
            unselectable="unselectable"
          >
            选择压力分类
          </div>
        </div>
        <span
          className="ant-select-arrow"
          onClick={[Function]}
          style={
            Object {
              "WebkitUserSelect": "none",
              "userSelect": "none",
            }
          }
          unselectable="unselectable"
        >
          <b />
        </span>
      </div>
    </div>
  </div>
  <div
    className="machine-input-flex"
  >
    <div
      className="machine-input-left"
    >
      燃料分类
    </div>
    <div
      className="ant-select ant-select-enabled"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={undefined}
      onMouseLeave={undefined}
      onTouchStart={[Function]}
      style={
        Object {
          "width": 400,
        }
      }
    >
      <div
        aria-autocomplete="list"
        aria-expanded={false}
        aria-haspopup="true"
        className="ant-select-selection
                  ant-select-selection--single"
        onKeyDown={[Function]}
        role="combobox"
        tabIndex={0}
      >
        <div
          className="ant-select-selection__rendered"
        >
          <div
            className="ant-select-selection__placeholder"
            onClick={[Function]}
            onMouseDown={[Function]}
            style={
              Object {
                "WebkitUserSelect": "none",
                "display": "block",
                "userSelect": "none",
              }
            }
            unselectable="unselectable"
          >
            选择燃料分类
          </div>
        </div>
        <span
          className="ant-select-arrow"
          onClick={[Function]}
          style={
            Object {
              "WebkitUserSelect": "none",
              "userSelect": "none",
            }
          }
          unselectable="unselectable"
        >
          <b />
        </span>
      </div>
    </div>
  </div>
  <div
    className="machine-input-flex"
  >
    <div
      className="machine-input-left"
    >
      排量分类
    </div>
    <div
      className="ant-select ant-select-enabled"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={undefined}
      onMouseLeave={undefined}
      onTouchStart={[Function]}
      style={
        Object {
          "width": 400,
        }
      }
    >
      <div
        aria-autocomplete="list"
        aria-expanded={false}
        aria-haspopup="true"
        className="ant-select-selection
                  ant-select-selection--single"
        onKeyDown={[Function]}
        role="combobox"
        tabIndex={0}
      >
        <div
          className="ant-select-selection__rendered"
        >
          <div
            className="ant-select-selection__placeholder"
            onClick={[Function]}
            onMouseDown={[Function]}
            style={
              Object {
                "WebkitUserSelect": "none",
                "display": "block",
                "userSelect": "none",
              }
            }
            unselectable="unselectable"
          >
            选择排量分类
          </div>
        </div>
        <span
          className="ant-select-arrow"
          onClick={[Function]}
          style={
            Object {
              "WebkitUserSelect": "none",
              "userSelect": "none",
            }
          }
          unselectable="unselectable"
        >
          <b />
        </span>
      </div>
    </div>
  </div>
  <div
    className="machine-input-flex"
  >
    <div
      className="machine-input-left"
    >
      设备所在地
    </div>
    <input
      className="ant-input machine-input-right"
      data-id="address"
      data-path="MACHINE"
      disabled={false}
      onChange={[Function]}
      onKeyDown={[Function]}
      placeholder="输入设备所在地"
      style={
        Object {
          "width": 400,
        }
      }
      type="text"
      value=""
    />
  </div>
  <div
    className="machine-input-flexs"
  >
    <div
      className="machine-input-left"
    >
      设备描述
    </div>
    <textarea
      className="ant-input"
      data-id="description"
      data-path="MACHINE"
      onChange={[Function]}
      onKeyDown={[Function]}
      placeholder="输入设备描述"
      rows={4}
      style={
        Object {
          "width": 400,
        }
      }
      value={undefined}
    />
  </div>
  <div>
    <div
      className="machine-input-left machine-input-flexs"
    >
      上传设备图片
    </div>
    <div
      className="machine-input-upload"
    >
      <span
        className=""
      >
        <div
          className="ant-upload-list ant-upload-list-picture-card"
          style={undefined}
        />
        <div
          className="ant-upload ant-upload-select ant-upload-select-picture-card"
          style={
            Object {
              "display": "",
            }
          }
        >
          <span
            className="ant-upload"
            onClick={[Function]}
            onDragOver={[Function]}
            onDrop={[Function]}
            onKeyDown={[Function]}
            role="button"
            style={undefined}
            tabIndex="0"
          >
            <input
              accept=""
              multiple={true}
              onChange={[Function]}
              style={
                Object {
                  "display": "none",
                }
              }
              type="file"
            />
            <div>
              <i
                className="anticon anticon-plus"
              />
              <div
                className="ant-upload-text"
              >
                上传图片
              </div>
            </div>
          </span>
        </div>
      </span>
    </div>
  </div>
</div>
`;
